pkg/controller/argocd/role_test.go

```diff
@@ -37,20 +38,27 @@ func TestReconcileArgoCD_reconcileRole(t *testing.T) {
 	assert.DeepEqual(t, expectedRules, reconciledRole.Rules)
 }
 
-func TestGetClusterRole(t *testing.T) {
+func TestReconcileArgoCD_reconcileClusterRole(t *testing.T) {
 	logf.SetLogger(logf.ZapLogger(true))
 	a := makeTestArgoCD()
 	r := makeTestReconciler(t, a)
 
-	createClusterRoles(t, r.client)
-
-	serverClusterRole := "argocd-server"
-	clusterRole, err := r.getClusterRole(serverClusterRole)
+	workloadIdentifier := common.ArgoCDApplicationControllerComponent
+	expectedRules := policyRuleForApplicationControllerClusterRole()
+	_, err := r.reconcileClusterRole(workloadIdentifier, expectedRules, a)
 	assertNoError(t, err)
-	assert.Equal(t, serverClusterRole, clusterRole.Name)
 
-	controllerClusterRole := "argocd-application-controller"
-	clusterRole, err = r.getClusterRole(controllerClusterRole)
-	assertNoError(t, err)
-	assert.Equal(t, controllerClusterRole, clusterRole.Name)
+	expectedName := fmt.Sprintf("%s-%s", a.Name, workloadIdentifier)
+	reconciledClusterRole := &v1.ClusterRole{}
+	assertNoError(t, r.client.Get(context.TODO(), types.NamespacedName{Name: expectedName}, reconciledClusterRole))
+	assert.DeepEqual(t, expectedRules, reconciledClusterRole.Rules)
+
+	// undersirable change.
```

Comment:
typo

```suggestion
	// undesirable change.
```

But, maybe it could really explain what it's changing/why it's changing it?